server/app/service/ScalaTestRunner.scala

```diff
@@ -18,6 +24,22 @@ object ScalaTestRunner {
   val traitDefPattern = """trait\s*([\w\$]*)""".r
   val defaultImports = "import org.scalatest._"
 
+  def apply(suiteClass: Class[Suite], solutionTrait: Class[AnyRef])(solution: String)(implicit s: Scheduler): Observable[Line] = {
```

Comment:
Moreover, second method would need reactive stuff as well. I mean:
def execSuite(solution: String, suite: String): String 

It will be the main method to run user-tests we are retrieving from database.